simpleviewinc/keg-components
# Zen 314 fix use theme path

**Ticket**: [ZEN-314](https://jira.simpleviewtools.com/browse/ZEN-314)
**Docker Image** [docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:zen-314-fix-usethemepath](docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:zen-314-fix-usethemepath)

## Context

* The `useThemePath` hook was throwing an error due to the `useMemo` being called at the wrong time in some cases
* When developing for both `keg-components` and `re-theme`, we would have to manually copy the re-theme changes from the build folder to the `keg-components/@simeplviewinc/re-theme/build` folder.
  * The storybook setup should allow watching the re-theme build folder, and restart when there is a change 

## Goal
* Ensure the `useThemePath` hook does not throw when being used
* Allow storybook to restart webpack when the re-theme build folder updates
* Add `nodemon` to watch the config folder
  * When a config file is change, storybook automatically is restarted
  * We no longer have to restart storybook manually when a config file has changed
  * Also allows restarts within the docker container without needing to re-start the container

## Updates

* `src/hooks/useThemePath.js`
  * Rewrote the `useThemePath` hook. It's now super simplified!
* `package.json`
  * Added `nodemon` for automatic server restarts
* `configs/webpack.config.js`
  * Added settings

## Testing

* **Setup**
  * Pull down and run the docker image
    * `keg dpg run docker.pkg.github.com/simpleviewinc/keg-packages/keg-components:zen-314-fix-usethemepath`
    * Once it's running, it should be accessable in the browser
* **Test 1**
  * Navigate to [components.kegdev.xyz](components.kegdev.xyz)
    * Select the contained button component
      * Ensure the styles render correctly
        * Check `default` / `hover` / `active` states
    * Adjust the screen size of the browser window
      * Ensure the no errors are thrown in the inspector
* **Test 2**
  * Leave the `keg-components` container running, and open a new terminal window
  * Connect to the running container
    * `keg d att components`
      * If it can't find the correct container, it will ask you to select one.
      * Enter the number of the running container
        * It should be called `package-keg-components-zen-314-fix-usethemepath`
  * Once connected, navigate to the re-theme directory in node modules
    * `cd node_modules/@simpleviewinc/re-theme`
  * Start the re-theme dev server
    * `yarn dev`
  * Once re-theme builds,
    * Ensure [components.kegdev.xyz](components.kegdev.xyz) reloads in your browser, and in the terminal window
* **Test 3**
  * If you're not happy with the auto-updating from the initial build, do this test
  * Make a change to one of the files in the `re-theme/src` directly
    * You have to do this manually because a mutagen sync is not setup
    * Use nano or vim to add a `console.log` to one of the `re-theme` files
  * Ensure `re-theme` rebuilds after making the change
  * Ensure [components.kegdev.xyz](components.kegdev.xyz) reloads with the changes from `re-theme`
